core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -158,14 +156,14 @@ object Partition extends KafkaMetricsGroup {
  * Data structure that represents a topic partition. The leader maintains the AR, ISR, CUR, RAR
  */
 class Partition(val topicPartition: TopicPartition,
```

Comment:
Did you mean to remove this `val` as well?

Reply:
This one is needed. Having a `val` in the type constructor generates a read-only method: `def topicPartition: TopicPartition`. This property (read-only method) is needed by other clients to this type.